src/Microsoft.TemplateEngine.Cli/Commands/InstantiateCommand.Help.cs

```diff
@@ -0,0 +1,453 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System.CommandLine;
+using System.CommandLine.Help;
+using System.CommandLine.Parsing;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Cli.Extensions;
+using Microsoft.TemplateEngine.Cli.TabularOutput;
+using Microsoft.TemplateEngine.Edge.Settings;
+using Microsoft.TemplateEngine.Utils;
+
+namespace Microsoft.TemplateEngine.Cli.Commands
+{
+    internal partial class InstantiateCommand
+    {
+        private const string Indent = "  ";
+
+        public IEnumerable<HelpSectionDelegate> CustomHelpLayout()
+        {
+            yield return (context) => WriteHelp(context, context.ParseResult);
+        }
+
+        public void WriteHelp(HelpContext context, ParseResult parseResult)
+        {
+            InstantiateCommandArgs instantiateCommandArgs = new InstantiateCommandArgs(this, parseResult);
+            if (string.IsNullOrWhiteSpace(instantiateCommandArgs.ShortName))
+            {
+                WriteCustomInstantiateHelp(context);
+                return;
+            }
+
+            IEngineEnvironmentSettings environmentSettings = GetEnvironmentSettingsFromArgs(instantiateCommandArgs);
+
+            using TemplatePackageManager templatePackageManager = new TemplatePackageManager(environmentSettings);
+            HostSpecificDataLoader hostSpecificDataLoader = new HostSpecificDataLoader(environmentSettings);
+
+            //TODO: consider use cache only for help
```

Comment:
https://github.com/dotnet/templating/issues/2191: consider creating TemplatePackageManager.GetTemplates version which uses cache only for tab completion and help